### ISSUE TITLE
Isolate GUI editor from IDE

### ISSUE THREAD

**Issue by albfan** [albfan/jmeld#18]
JMeld comes with group layout forms

https://docs.oracle.com/javase/tutorial/uiswing/layout/group.html

which is tied to Netbeans https://netbeans.org/kb/docs/java/quickstart-gui.html

Every little change made, generates a lot of noise which hides the change behind.. See https://github.com/albfan/jmeld/commit/4fc9004106b72aac2011146e98bf57898f736d9a as an example, I want to avoid that.

Other solutions are tied to each IDE too:

https://www.jetbrains.com/idea/help/designing-gui-major-steps.html

Seems there is no "rosseta stone" for java forms. Best shot is [jformdesigner](http://www.formdev.com/) but is privative and it converts metadata form to... another closed form format.

Other good solutions are:

http://www.jgoodies.com/products/swing-suite/
http://www.miglayout.com/

The objetive is to design complex GUI layouts with a compact and robust system, wether it is command line, visual, or compiled from metadata.

**Comment by opticyclic** [comment]
If you are going to rewrite the GUI, maybe do it in JavaFX or go crazy and use [Electron! ](http://electron.atom.io/)

**Comment by albfan** [comment]
Indeed, plans are to offer core as a REST api, so electron implementations should be easy to do.

**Comment by albfan** [comment]
Although not apparently related, working on https://github.com/albfan/SocketTest/issues/3 gives the tips to finish this issue (as it is a more simple program, easy to translate to MVC)

**Comment by albfan** [comment]
Have to take a look at fxml editor from javafx https://picodotdev.github.io/blog-bitix/2015/10/javafx-scene-builder-editor-para-crear-archivos-fxml/

**Comment by albfan** [comment]
I did a quick test and fxml works correctly.

https://github.com/albfan/javafxTest
